### ISSUE TITLE
Execute Filebeat from a service user account to avoid security problems

### ISSUE THREAD

**Issue by vmcano** [elastic/filebeat#148]
The init script should be modified to execute Filebeat from a service user account to avoid security problems. If there is a bug in Filebeat that allows to take control over the service and it was executed with root privileges, the attacker will have root access in our system.
A possible modification to do this could be:
````
    daemonopts="--user $service_user --pidfile $pidfile"
                  .
                  .
                  .

    touch $pidfile || return 1
    chown elk $pidfile || return 1
    chgrp elk $pidfile || return 1
    chmod g+w $pidfile || return 1

    daemon $daemonopts $wrapper $wrapperopts -- $agent $args
````

Another modification would be creating a user account when installing Filebeat.

**Comment by andrewkroh** [comment]
I agree with applying the PoLP, but wouldn't this cause an issue with filebeat reading log files owned by other services?

**Comment by andrewkroh** [comment]
BTW, the [init scripts live here](https://github.com/elastic/beats-packer/tree/master/platforms) if you are interested in viewing them for other platforms or opening a pull request.

**Comment by vmcano** [comment]
Well, as you say, it would lead to problems reading log files if permissions aren't correctly set, so it may be better to leave it as it is.
If someone is more worried about security, he can change the script locally.

**Comment by iammichiel** [comment]
@vmcano If that is your point of view, that means that filebeat is not encouraging best practices of linux security. It should be the other way round. Make the best-practice a default and leave a fallback for an easier but less-secure solution.

**Comment by manuel-sousa** [comment]
I agree with @iammichiel , I'd rather have a different user for filebeats as I used to have with logstash. One trick that made my life easier was ensure that all logs had group read enabled for adm group and add the logstash user to adm.

Additionally, in my opinion adm membership should be included for filebeat if it runs as service account instead of root as it's main purpose is reading log files.

**Comment by lhoss** [comment]
+1 for running under special service user
+1 also for creating a specific filebeat OS user  (I'm deploying filbert (with ansible) on many mesos/spark slave nodes, but without logstash package being installed, so no 'logstash' user available)

**Comment by cdelgadob** [comment]
Any news about this enhancement? 

What is the recommended procedure to install Filebeat in enterprise environments?

**Comment by mikeforbes** [comment]
+1 for this.

I'm having to modify the default init script (supplied via debian package) to allow filebeat to run as a non-root user. (as well as creating a filebeat user & group, adding to 'adm' group to read logfiles)

It would be great if this was a feature.

**Comment by ruflin** [comment]
@mikeforbes Can you share the modifications you make?

**Comment by mikeforbes** [comment]
@ruflin Surel here it is:

https://gist.github.com/mikeforbes/748895127b2f75a9841ae66d4c89e824

Be sure to create the 'filebeat' user as specified in the comments of the init script there and of course you'll need to make sure the user is in the appropriate groups to read the log files you're sending to filebeat.
In my case adding to the 'adm' group was sufficient.

**Comment by ruflin** [comment]
@mikeforbes Thanks for sharing.
